README.md

```diff
@@ -72,32 +121,49 @@ func sayHelloToScratchy() {
 }
 ```
 
-* Instead of doing that, ask the student if they can recognize the common pattern here. What is the only difference between these three functions. Use this as a way to jump into functions with arguments:
+Add those to your playground file. You probably noticed as you typed those in that they're pretty repetitive. What do they all have in common?
+
+Save for the name of your cat friend, they all have the same format: printing the string "Hello &lt;cat friend&gt;, why do you sleep so much?" Doesn't that defeat the purpose of having code in a function so you only have to type it once?
+
+Luckily, you can generalize functions using _arguments_. You've already seen arguments in action: You've passed different strings to the `print()` function to print different things. There is nothing special about Swift's `print()` function; your own functions, too, can take arguments, which will slightly change how they work. Let's take a look at that.
+
+Remember when you created `sayHello()`, and you added an empty set of parentheses after the function name? You can specify an argument in those parentheses instead. That argument consists of both a name and the argument's data type (you remember learning about data types, right?). You can then pass in a different value in place of that argument when you call the function. Inside the function's body, you can refer to that value using the name in the argument list.
+
+Let's try to generalize `sayHelloToGarfield()`, etc., a bit, so they consist of just _one_ function that uses an argument. Here's how you would define a more general function, `sayHelloToCat()`:
 
 ```swift
-func sayHelloToCat(kittenName: String) {
-    print("Hello, \(kittenName), why do you sleep so much?")
+func sayHelloToCat(catName: String) {
+    print("Hello, \(catName), why do you sleep so much?")
 }
+```
 
-func sayHelloTDog(puppyName: String) {
-    print("Hello, \(puppyName), want to go sniff that rock together?")
-}
+Enter the above code into your playground file. Take a look at what you did: You defined a function (using the `func` keyword) called `sayHelloToCat`. Like `sayHello`, you opened with a set of parentheses. But instead of being empty, you declared an argument called `catName` of type `String`. Then, you wrote a function body consisting of a call to `print()`. Notice that the string you are printing interpolates the argument, `catName`. `catName` is a constant that you can use anywhere inside the `sayHelloToCat()` function. You can think of it as being a constant declared inside of the function body, just like `greeting` was declared inside of `sayHello()`. Like `greeting`, `catName` is visible only within the _scope_ of `sayHelloToCat`, so you cannot refer to it outside of the function.
+
+You can call `sayHelloToCat()` just like you called `sayHello()`, except you have to pass an argument to `sayHelloToCat()`. You can simply pass a string:
 
+```swift
+sayHelloToCat("Mittens")
 ```
```

Comment:
I like adding comments that show the student what it is that will be printed to the console. Your sentence after this code snippet does tell them, which can stay. Having both (comment below function call and your sentence after) is fine here.

sayHelloToCat("Mittens")
// prints "Hello, Mittens, why do you sleep so much?"